### PR TITLE
Making GroupNorm and LayerNorm more compatible with X10

### PR DESCRIPTION
**Pull request by BradLarson**
GroupNorm's local `epsilon` Tensor was being created on the default TF eager device, which led to a crash on X10 devices. This creates an appropriately placed Tensor for that value.

Additionally, wrapping the Tensor creation in `withoutDerivative(at:)` within both GroupNorm and LayerNorm prevents that value from being scalarized in the backwards pass, which would cut the X10 trace at that point. We've found that this has a significant performance impact within BatchNorm, so I've prevented that here.

This should fix issue #1031.

### CODE REVIEW

**Comment by mikowals** [comment]
Could there be comments added to explain why this is necessary?  

I tripped up earlier removing the code like this from batch norm and think it takes pretty thorough understanding of X10 and autodiff to reason why these are needed.

**Comment by BradLarson** [comment]
@mikowals - Good point, we didn't even know this was an issue until we saw the traces being cut here. I've added notes around the three places that I know will require this right now. There are other locations using `withoutDerivative(at:)` here where I believe it should be safe to remove that without impacting traces, and I've left those alone. They will need to have traces examined before and after to make sure, but if there's no change in the backwards pass we should be good to remove those.
